src/components/general/DatePicker.js

```diff
@@ -15,99 +16,76 @@ class DatePicker extends Component {
-			focused: date.length === 3
+			yearArr: Array.from({ length: 20 }, (v, k) => new Date().getFullYear() + k) // We only need about 20 years ahead... right?
 		};
 	}
 
-		static propTypes = {
-			value: PropTypes.object,
-			placeholder: PropTypes.string.isRequired,
-			onSelect: PropTypes.func.isRequired
-		}
+	static propTypes = {
+		value: PropTypes.object,
+		placeholder: PropTypes.string.isRequired,
+		onSelect: PropTypes.func.isRequired
+	}
 
-		prepend0(item) {
-			if (item < 10)
-				return "0" + item;
+	// Returns an array of days in the current state's month when called
+	dayArr() {
+		const {
+			month,
+			day,
+			year
+		} = this.state;
 
-			return item;
-		}
+		// Return the correct array of days based on the selected month
+		// Defining # of days for each month is easier than whatever monstrosity was here before
+		const daysInMonth = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
 
-		leapYear(year) {
-			return year % 4 == 0 && year % 100 != 0 || year % 400 == 0;
-		}
+		// If the state's day is more than the max days in a month, set the day to the max day
+		const maxDayCheck = (maxDay) => { if (day > maxDay) this.setState({ day: maxDay }); };
 
-		update(item) {
-			const {
-				month,
-				day,
-				year
-			} = item;
-			this.props.onSelect(`${year}-${month}-${day}`);
+		if (parseInt(month) == 2 && this.isLeapYear(year)) {
+			maxDayCheck(29);
+			return Array.from({ length: 29 }, (v, k) => k + 1);
 		}
-
-		clickActionMonth(item) {
-			const {
-				day,
-				year
-			} = this.state;
-
-			item = this.prepend0(item);
-
-			this.setState({ month: item });
-			const month30 = [false, false, false, true, false, true,
-				false, false, true, false, true, false];
-
-			if (item === 2)
-				if (year != 0 && this.leapYear(year)) {
-					this.setState({ dayArr: Array.from({ length: 29 }, (v, k) => k + 1) });
-					if (day > 29) this.setState({ day: 29 });
-				}
-				else {
-					this.setState({ dayArr: Array.from({ length: 28 }, (v, k) => k + 1) });
-					if (day > 28) this.setState({ day: 28 });
-				}
-			if (month30[item]) {
-				this.setState({ dayArr: Array.from({ length: 30 }, (v, k) => k + 1) });
-				if (day > 30) this.setState({ day: 30 });
-			}
-			else if (item !== 2) {
-				this.setState({ dayArr: Array.from({ length: 31 }, (v, k) => k + 1) });
-			}
-			if (day !== "" && year !== "")
-				this.update({ day: day, month: item, year: year });
+		else {
+			maxDayCheck(daysInMonth[parseInt(month) - 1]);
+			return Array.from({ length: daysInMonth[parseInt(month) - 1] }, (v, k) => k + 1);
 		}
-		clickActionDay(item) {
-			const {
-				month,
-				year
-			} = this.state;
+	}
 
-			item = this.prepend0(item);
-			this.setState({ day: item });
+	// Returns true if the input year is a leap year
+	isLeapYear(year) {
+		return year % 4 == 0 && year % 100 != 0 || year % 400 == 0;
+	}
 
-			if (month !== "" && year !== "")
-				this.update({ day: item, month: month, year: year });
-		}
+	update(item) {
```

Comment:
This should have a name that reflects what it's doing. `updateParent` or `submitDate` or something of that nature.